### PR TITLE
Documentation Update of 2020.01

### PR DESCRIPTION
**Pull request by myungjoo**
We have been progressed a lot. Update the documentation accordingly.

Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>

### CODE REVIEW

File: Documentation/coding-convention.md [hunk 1]
```diff
@@ -1,16 +1,79 @@
...
+
+## Directory structure of nnstreamer.git
+
+- **api**: API definitions and implementations
+    - **android**: Android APIs
+    - **capi**: C-APIs (Tizen and others)
+    - Note that .NET APIs are located at https://github.com/Samsung/TizenFX along with other Tizen .NET APIs
+- **debian**: Debian/Ubuntu packaging files
+- **Documentation**: Documentations
+- **ext/nnstreamer**: NNStreamer plugins and subplugins that depend on optional or non-stndard external packages. Components in this directory can be built and included optionally.
```
Comment (jaeyun-jung) [inline Documentation/coding-convention.md]: typo non-stndard

File: Documentation/contributing.md [hunk 2]
```diff
@@ -6,6 +6,8 @@ Consistent code conventions are important for several reasons:
 * To make it easy to debug the code, with both a system call tracer and GNU debuggers. It should be easy to set breakpoints, view locals, and display and view data structures.
 * To attempt to improve code quality through consistency, and requiring patterns that are less likely to result in bugs either initially, or after code modification.
 
+For more information, plrease refer to [coding-convention.md](coding-convention.md).
```
Comment (dongju-chae) [inline Documentation/contributing.md]: typo: plrease

File: Documentation/coding-convention.md [hunk 3]
```diff
@@ -1,16 +1,79 @@
...
+- **jni**: Android/Java build scripts.
+- **nnstreamer\_example**: Example custom filters required by test cases. In the old days, we used to have all nnstreamer examples here. Most of such examples are moved to [Example git](https://github.com/nnsuite/nnstreamer-example) except for those who required by test cases.
+- **packaging**: Tizen RPM build scripts. OpenSUSE/Redhat Linux may reuse this.
+- **tests**: Unit test cases. We have SSAT test cases and GTEST test cases. There are a lot of subdirectories, which are groups of unit test cases.
+- **tools**: Various developmental tools and scripts of NNStreamer.
+
+## Related git repositories in nnsuite
+
+- [NNStreamer Examplie Applications \& Documents](https://github.com/nnsuite/nnstreamer-example)
+- [TAOS-CI, CI Service for On-Device AI Systems](https:?/github.com/nnsuite/TAOS-CI)
```
Comment (jaeyun-jung) [inline Documentation/coding-convention.md]: link error : https:?/g

File: Documentation/coding-convention.md [hunk 4]
```diff
@@ -1,16 +1,79 @@
...
+- [NNStreamer ROS (Robot OS) Support](https://github.com/nnsuite/nnstreamer-ros)
+- [NNStreamer Android Build Resource](https://github.com/nnsuite/nnstreamer-android-resource): additional files required by Android builds.
+- [ORC, aarch64 support](https://github.com/nnsuite/orc): we are going to support aarch64-ORC to accelerate transform operations of nnstreamer in aarch64 devices.
+- [NNStreamer-Edge](https://github.com/nnsuite/nnstreamer-edge): WIP
+- [NNStreamer Yocto/OpenEmbedded Layer](https://github.com/nnsuite/meta-neural-network): refer to [Opembedded layer page](https://layers.openembedded.org/layerindex/branch/master/layer/meta-neural-network/)
```
Comment (jaeyun-jung) [inline Documentation/coding-convention.md]: typo [Opembedded layer page]

File: Documentation/contributing.md [hunk 5]
```diff
@@ -6,6 +6,8 @@ Consistent code conventions are important for several reasons:
 * To make it easy to debug the code, with both a system call tracer and GNU debuggers. It should be easy to set breakpoints, view locals, and display and view data structures.
 * To attempt to improve code quality through consistency, and requiring patterns that are less likely to result in bugs either initially, or after code modification.
```
Comment (jaeyun-jung) [inline Documentation/contributing.md]: and requiring patterns **that** are less likely

File: Documentation/contributing.md [hunk 6]
```diff
@@ -21,7 +23,21 @@ The submitter has the first responsibility of keeping the created PR clean and n
 * If your functions or structs/classes are going to be accessed by other modules or NNStreamer users, provide full descriptions of all entries with Doxygen.
 * Passing all the tests of TAOS-CI is a neccesary condition, but not a satisfying condition.
```
Comment (jaeyun-jung) [inline Documentation/contributing.md]: typo neccesary

File: Documentation/contributing.md [hunk 7]
```diff
@@ -21,7 +23,21 @@ The submitter has the first responsibility of keeping the created PR clean and n
...
+A PR is required to meet the following criteria.
+* It has passed all the tests defined for TAOS-CI.
+    - This includes unit tests and integration tests in various platforms and different static analysis tools.
+    - Note that one of the tests includes the "Signed-off-by" check, which means that the author has agreed with [Code of Conduct](../CODE_OF_CONDUCT.md). You may need to refer to later section.
+* At least TWO official reviewers have approved the PR.
+    - This does not guarantee accepts. This is a necessary condition, not sufficient.
+    - If the PR touches sensitive codes or may affect wide ranges of components, reviewers will wait for other reviewers to back them up.
+    - If the PR is messy, you will need to wait indefinitely to get reviews.
+        - Apply general rules of git commits and common senses.
+        - Do not write a lengthy commit. Apply a single commit per PR if you are new to the community. Have a single topic per commit. Provide enough background information and referneces. And so on.
```
Comment (jaeyun-jung) [inline Documentation/contributing.md]: typo referneces.

File: Documentation/coding-convention.md (hunk 1 above)
Comment (myungjoo) [reply Documentation/coding-convention.md]: Fixed.

File: Documentation/contributing.md (hunk 2 above)
Comment (myungjoo) [reply Documentation/contributing.md]: Fixed

File: Documentation/coding-convention.md (hunk 3 above)
Comment (myungjoo) [reply Documentation/coding-convention.md]: Fixed

File: Documentation/coding-convention.md (hunk 4 above)
Comment (myungjoo) [reply Documentation/coding-convention.md]: Fixed.

File: Documentation/contributing.md (hunk 5 above)
Comment (myungjoo) [reply Documentation/contributing.md]: It's not an error.

File: Documentation/contributing.md (hunk 6 above)
Comment (myungjoo) [reply Documentation/contributing.md]: Fixed.

File: Documentation/contributing.md (hunk 7 above)
Comment (myungjoo) [reply Documentation/contributing.md]: Fixed

**Pull request merged**